src/network.cc

```diff
@@ -161,7 +165,21 @@ QWebSocket *webSocket =
 #line 126 "network.nw"
   QUrl url(luaL_checkstring(L, 2));
   url.setPort(lua_tointegerx(L, 3, NULL));
-  webSocket->open(url);
+
+  QEventLoop loop;
+  #line 170 "network.nw"
+
+  // Create connection when websocket connected 
+  QObject::connect(webSocket, SIGNAL(connected()), &loop, SLOT(quit()));
+  QObject::connect(webSocket, SIGNAL(disconnected()), &loop, SLOT(quit()));
+
+  // Wait until socket connection is established
+  while (webSocket->state() != QAbstractSocket::ConnectedState) {
+    webSocket->open(url);
+    loop.exec();
+    usleep(100);
+  }
+  
```

Comment:
@VVeremjova,
1. Maybe avoid `loop` with connecting [`error signal`](http://doc.qt.io/qt-5/qwebsocket.html#error-1) to `open slot`?
2. Could we use the root loop of qApp instead of creation own one? Or we trying do let to pressed all other events?